multiqc/modules/humid/neighbours.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from multiqc.plots import linegraph
+
+log = logging.getLogger(__name__)
+
+
+def parse_reports(self):
+    # To store the summary data
+    self.neighbours = dict()
+
+    # Parse the output files
+    parse_log_files(self)
+
+    # Remove filtered samples
+    self.neighbours = self.ignore_samples(self.neighbours)
+
+    log.info(f"Found {len(self.neighbours)} reports")
+    self.write_data_file(self.neighbours, "multiqc_humid_neighbours")
+
+    add_to_humid_section(self)
+
+
+def parse_log_files(self):
+    for f in self.find_log_files("humid/neighbours"):
+        # There is no sample name in the log, so we use the root of the
+        # file as sample name (since the filename is always stats.dat
+        s_name = self.clean_s_name(f["root"], f)
+
+        # process the file content
+        d = {}
+        for line in f["contents_lines"]:
+            nr_neighbours, count = line.strip("\n").split(" ")
+            d[int(nr_neighbours)] = int(count)
+
+        # Superfluous function call to confirm that it is used in this module
+        # Replace None with actual version if it is available
+        self.add_software_version(None)
+
+        # Got this far, data must be good
+        if s_name in self.neighbours:
+            log.debug("Duplicate sample name found! Overwriting: {s_name}")
+        self.neighbours[s_name] = d
+        self.add_data_source(f, s_name)
+
+
+def add_to_humid_section(self):
+    # Figure configuration
+    plot_config = {
+        "id": "humid-neighbours",
+        "title": "HUMID: Neighbour statistics",
+        "ylab": "Number of sequences",
+        "xlab": "Number of neighbours",
+        "logswitch": True,
+        "logswitch_active": True,
+        "hide_zero_cats": False,
+    }
+    self.add_section(
```

Comment:
If neigh.dat is not found, an empty plot will be generated. Would be nice to check if `self.neighbours` is non-empty before the `self.add_section` call

Reply:
Each submodule separately searches for their associated log file, so if a file is completely missing it will not be included in the report.
The tool for this module (HUMID) will never generate empty statistics files.

Reply:
I did some more testing, and you can indeed add up with an empty section if there is only a single sample, and that sample cluster file has no data. I've fixed this in the latest commit, thanks!